### PR TITLE
feat: Add full and partial export functionality

### LINKED ISSUE #29275: Task: full page download feature with partial import export.
Currently we give users to select what they want to export from the current app.
But for scenarios in which users want to reuse the whole page from one app to another, full page export would be helpful

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This pull request adds the functionality to export both the full page and selected entities from the page. It includes the following changes:

- Added full page export functionality

- Refactored partial export functionality

- Added partialExportUtils and useEntitesToExport files

These changes allow users to export the entire page or select specific entities to export. The export functionality is now more efficient and user-friendly.

#### PR fixes following issue(s)
Fixes #29275

#### Media

https://github.com/appsmithorg/appsmith/assets/6761673/f04c4094-a5f5-41da-945f-14820f09cb22

- New feature (non-breaking change which adds functionality)
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Enhanced the export modal with clearer instructions for users selecting entities to export.
  - Introduced a new "Export all" feature for exporting all entities from a page.

- **Bug Fixes**
  - Corrected the component name from `PartiaExportModel` to `PartiaExportModal` for consistency.

- **Refactor**
  - Streamlined the export process with a new custom hook `useEntitesToExport`.
  - Improved the selection logic for exporting widgets with new utility functions.

- **Style**
  - Updated UI elements in the export modal for a better user experience.

- **Documentation**
  - Updated text in the export modal to guide users more effectively.

- **Chores**
  - Removed unused components and functions related to the export feature.

### CODE REVIEW

File: app/client/src/components/editorComponents/PartialImportExport/PartialExportModal/partialExportUtils.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      return result;
+    },
+    [[], []],
+  );
+  const allWidgets =
+    canvasWidgets?.children?.map((child) => child.widgetId) || [];
+
+  const allCustomJSLibs =
+    customJsLibraries.map((jsLib) => jsLib.id || "") || [];
+  const allDatasources = appWideDS.map((ds) => ds.id) || [];
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/components/editorComponents/PartialImportExport/PartialExportModal/partialExportUtils.ts]: Done

File: app/client/src/components/editorComponents/PartialImportExport/PartialExportModal/index.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import {
   PARTIAL_IMPORT_EXPORT,
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/components/editorComponents/PartialImportExport/PartialExportModal/index.tsx:2]: Done.

**Review by jacquesikot — DISMISSED** [review]

**Comment by rahulbarwal** [comment]
Not needed anymore.

**Pull request closed without merging**